timely_beliefs/beliefs/classes.py

```diff
@@ -458,7 +465,13 @@ def __init__(  # noqa: C901 todo: refactor, e.g. by detecting initialization met
                         self[col] = None
                     self.set_index(indices, inplace=True)
                     self.sensor = sensor
-                    self.event_resolution = sensor.event_resolution if sensor else None
```

Comment:
This init method was indeed up for significant refactoring. It remains large, but it should now be a lot clearer to read, especially because I got rid of all the return statements that provide alternative exits.